core/kazoo_documents-1.0.0/src/kzd_alert.erl

```diff
@@ -0,0 +1,254 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2015, 2600Hz
+%%% @doc
+%%% Alert document
+%%% @end
+%%% @contributors
+%%%   Peter Defebvre
+%%%-------------------------------------------------------------------
+-module(kzd_alert).
+
+-export([new/0
+         ,type/0
+         ,id/1
+         ,fetch/1
+        ]).
+
+-export([title/0, title/1, title/2, set_title/2
+         ,category/0 ,category/1, category/2, set_category/2
+         ,message/0, message/1, message/2, set_message/2
+         ,metadata/0, metadata/1, metadata/2, set_metadata/2
+         ,level/0, level/1, level/2, set_level/2
+         ,from/0, from/1, from/2, set_from/2
+         ,to/0, to/1, to/2, set_to/2
+         ,expiration_date/0, expiration_date/1, expiration_date/2, set_expiration_date/2
+         ,expired/1
+        ]).
+
+-define(ID, <<"_id">>).
+-define(PVT_TYPE, <<"alert">>).
+
+-define(TITLE, <<"title">>).
+-define(CATEGORY, <<"category">>).
+-define(MESSAGE, <<"message">>).
+-define(METADATA, <<"metadata">>).
+-define(LEVEL, <<"level">>).
+-define(FROM, <<"from">>).
+-define(TO, <<"to">>).
+-define(EXPIRATION_DATE, <<"expiration_date">>).
+
+-include("kz_documents.hrl").
+
+-type doc() :: wh_json:object().
+-export_type([doc/0]).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec new() -> doc().
+new() ->
+    wh_doc:set_type(wh_json:new(), ?PVT_TYPE).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec type() -> ne_binary().
+type() -> ?PVT_TYPE.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec id(doc()) -> api_binary().
+id(JObj) ->
+    wh_doc:id(JObj).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec fetch(api_binary()) -> {'ok', doc()} |
+                            {'error', any()}.
+fetch('undefined') ->
+    {'error', 'invalid_db_name'};
+fetch(<<_/binary>> = AlertId) ->
+    couch_mgr:open_cache_doc(?WH_ALERTS_DB, AlertId).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec title() -> ne_binary().
+-spec title(doc()) -> api_binary().
```

Comment:
How can we help you write proper specs? I feel like your PRs too often require us to go through and tell you which ones are incorrect, which is unfair to our time. You have the tools and intelligence to do it yourself.

The first thing I do when I feel "done" is to dialyze the code I've made changes to:
```bash
kazoo$> dialyzer --plt .kazoo.plt core/kazoo_documents-1.0.0/ebin/kzd_alert.beam
```

Once I feel like its internally consistent, I dialyze with the modules referenced:

```bash
kazoo$> dialyzer --plt .kazoo.plt core/kazoo_documents-1.0.0/ebin/kzd_alert.beam core/whistle-1.0.0/ebin/wh_json.beam ...
```

If both passes go successfully, I'm feeling good. Dialyzer thinks things are both internally consistent and consistent with the modules I'm calling off to, so I can be reasonably confident I've wired things up right.

Does that make sense? Do you have questions about how all this works? Please ask if so; we all get to learn more that way.

Same goes for @lazedo @lfurrea @icehess @k-anderson @fenollp; if anyone has questions about how others work, how to do something more efficiently, how to take advantage of tooling to make life easier, etc, lets share those questions and ideas. Write engineering@ so we can all participate in getting better.